core/src/migrations/000068-profilePropertyCompositeUniqueIndex.ts

```diff
@@ -0,0 +1,65 @@
+import { config } from "actionhero";
+import { Op } from "sequelize";
+
+export default {
+  up: async function (migration, DataTypes) {
+    await migration.sequelize.transaction(async () => {
+      await migration.addColumn("profileProperties", "unique", {
+        type: DataTypes.BOOLEAN,
+        defaultValue: false,
+        allowNull: true,
+      });
+
+      const [properties] = await migration.sequelize.query(
+        "SELECT * FROM \"properties\" WHERE state = 'ready'"
+      );
+      const uniqueProperties = properties.filter(
+        (p) => p.unique === 1 || p.unique === "1" || p.unique === true
+      );
+
+      if (uniqueProperties.length > 0) {
+        await migration.sequelize.query(
+          `UPDATE "profileProperties" SET "unique"='${
+            uniqueProperties[0].unique
+          }' WHERE "propertyId" IN (${uniqueProperties
+            .map((p) => `'${p.id}'`)
+            .join(", ")})`
+        );
+      }
```

Comment:
This migration could be very slow for large datasets - it needs to update every row (profiles x number of properties)